src/Http/Request/RequestsExecutor.ts

```diff
@@ -162,123 +168,22 @@ export class RequestsExecutor {
-        .then((): BluebirdPromise<IChooseNodeResponse> => this
-        .chooseNodeForRequest(command)
-      );
-    }
-
-    try {
-      response = (command.isReadRequest)
-        ? this.chooseNodeForRead(command)
-        : this.chooseNodeForWrite(command);
-    } catch (error) {
-      return BluebirdPromise.reject(error as RavenException) as BluebirdPromise<IChooseNodeResponse>;
-    }
-
-    return BluebirdPromise.resolve(response) as BluebirdPromise<IChooseNodeResponse>;
-  }
-
-  protected chooseNodeForRead(command: RavenCommand): IChooseNodeResponse {
+  protected chooseNodeForRequest(command: RavenCommand): BluebirdPromise<ServerNode> {
     const topology: Topology = this._topology;
-    const leaderNode: ServerNode = topology.leaderNode;
-
-    switch (topology.readBehavior) {
-      case ReadBehaviors.LeaderOnly:
-        if (!command.isFailedWithNode(leaderNode)) {
-          return {node: leaderNode};
-        }
-
-        throw new RequestException(
-          'Leader node failed to make this request. The current read behavior is set to LeaderOnly'
-        );
-      case ReadBehaviors.RoundRobin:
-        let nonFailedNode: ServerNode;
-        let skippedNodes: ServerNode[] = [];
+    let nonFailedNode: ServerNode;
 
-        if ([leaderNode].concat(topology.nodes).some((node: ServerNode): boolean => {
-            const nonFailed = !node.isFailed && !command.isFailedWithNode(node);
+    if (!topology.nodes.some((node: ServerNode): boolean => {
+        const nonFailed = !node.isFailed && !command.isFailedWithNode(node);
 
-            nonFailed
-               ? (nonFailedNode = node)
-              : skippedNodes.push(node);
-
-            return nonFailed
-          })) {
-          return {node: nonFailedNode, skippedNodes: skippedNodes};
-        }
-
-        throw new RequestException(
-          'Tried all nodes in the cluster but failed getting a response'
-        );
-      case ReadBehaviors.LeaderWithFailoverWhenRequestTimeSlaThresholdIsReached:
-        if (!leaderNode.isFailed && !command.isFailedWithNode(leaderNode)
-          && leaderNode.isRateSurpassed(topology.sla)
-        ) {
-          return {node: leaderNode};
-        }
-
-        const nonFailedNodes: ServerNode[] = [leaderNode]
-          .concat(topology.nodes || [])
-          .filter((node: ServerNode): boolean => !node.isFailed)
-          .sort((node: ServerNode, anoterNode: ServerNode): number =>
-            node.ewma - anoterNode.ewma
-          );
-
-        if (nonFailedNodes.length > 0) {
-          return {node: nonFailedNodes[0], skippedNodes: nonFailedNodes.slice(1)};
-        }
-
-        throw new RequestException(
-          'Tried all nodes in the cluster but failed getting a response'
-        );
-      default:
-        throw new InvalidOperationException(
-          StringUtil.format('Invalid read behavior value: {readBehavior}', topology)
-        );
+        nonFailed && (nonFailedNode = node);
```

Comment:
Using `some()` and having side-effects like this in the callback function makes code a bit cryptic, don't you think? In general I never use not *pure* functions for callbacks like this. A regular for loop would be more readable here.